playground/frontend/playground_components/lib/src/services/analytics/analytics_service.dart

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'dart:async';
+import 'dart:collection';
+
+import 'package:get_it/get_it.dart';
+import 'package:meta/meta.dart';
+
+import 'events/abstract.dart';
+
+/// Abstract class to submit analytics events.
+abstract class AnalyticsService {
+  AnalyticsEvent? _lastEvent;
+  AnalyticsEvent? get lastEvent => _lastEvent;
+
+  final _defaultEventParameters = <String, dynamic>{};
+
+  /// The parameters sent with all events.
+  ///
+  /// Individual event parameters have higher priority on collisions.
+  Map<String, dynamic> get defaultEventParameters =>
+      UnmodifiableMapView(_defaultEventParameters);
+
+  set defaultEventParameters(Map<String, dynamic> newValue) {
+    _defaultEventParameters.clear();
+    _defaultEventParameters.addAll(newValue);
+  }
+
+  static AnalyticsService get() {
+    return GetIt.instance.get<AnalyticsService>();
```

Comment:
I don't see how GetIt is making it clear where the actual implementation exists.  I had to search quite a bit until I found the actual implementation.  Nonetheless, if we decide to keep and use GetIt, can we make good use of it and expand our test coverage as the package promises to make easier?

Reply:
It does not because its goal is to decouple usage from implementation.

A reader familiar with GetIt would find-in-files 'GetIt.instance.register' to get to implementations.

As for swapping in mocks in tests, GetIt can do that for unit tests because there we can have different initialization. But we aim to not add new mocks to unit tests because we plan to migrate to another mocking library in the next phase.

Instead we test analytics with integration tests (since we anyway test everything by clicking through), and there we do not have the control of implementations because the real-app implementations are loaded. Thus we check `BeamAnalyticsService.lastEvent`.

The only real benefit of GetIt that we use for now is to make `playground_components` unaware of the implementation without using Provider and global variables, which justifies it to us.